### DISCUSSION TITLE
Nautobot "key" secret type and device credentials

### DISCUSSION THREAD

**Discussion by ghost** [nautobot/nautobot#5813] [Q&A] [answered]
Hello Community !

Got devices with password-less connection type (ssh key): does the Nautobot "key" secret type can be used to connect to devices with ssh key (and without password) ?

I read the doc carefully but I couldn't find any information on this Nautobot secret type use.

Have done some tests with Nautobot 2.1.4, Device onboarding 3.0.1, Golden configuration 2.1.0, secret group with username, blank password and ssh private key file :
- Device onboarding: needs NAPALM args "use_keys" and "key_file" to be set on platform level
- Golden configuration: requires a no blank password (error "E1006: There was no password (redacted) preemptively failed.)

Maybe I missed something ?

Many thanks.
Regards.

**Comment by jdrew82** [comment, marked as answer]
At this time we don't support SSH keys for device connections. GitRepositories only support HTTPS connections and must support the username/personal access token in the URL. Connections with NAPALM are dependent upon what the driver supports but I believe most use username/password/secret combos.
